pkg/render/logstorage/components/elasticsearch_metrics.go

```
@@ -0,0 +1,151 @@
+package component
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/tigera/operator/pkg/render/common/secret"
+
+	"github.com/tigera/operator/pkg/ptr"
+	relasticsearch "github.com/tigera/operator/pkg/render/common/elasticsearch"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/render"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func ElasticsearchMetrics(
+	installation *operatorv1.InstallationSpec,
+	clusterConfig *relasticsearch.ClusterConfig,
+	esMetricsCredsSecret *corev1.Secret,
+	esCertSecret *corev1.Secret,
+	clusterDomain string,
+) render.Component {
+	return &elasticsearchMetrics{
+		installation:         installation,
+		clusterConfig:        clusterConfig,
+		esMetricsCredsSecret: esMetricsCredsSecret,
+		esCertSecret:         esCertSecret,
+		clusterDomain:        clusterDomain,
+	}
+}
+
+type elasticsearchMetrics struct {
+	installation         *operatorv1.InstallationSpec
+	esMetricsImage       string
+	esMetricsCredsSecret *corev1.Secret
+	esCertSecret         *corev1.Secret
+	clusterConfig        *relasticsearch.ClusterConfig
+	clusterDomain        string
+}
+
+func (e *elasticsearchMetrics) ResolveImages(is *operatorv1.ImageSet) error {
+	var err error
+	var errMsgs []string
+
+	reg := e.installation.Registry
+	path := e.installation.ImagePath
+
+	e.esMetricsImage, err = components.GetReference(components.ComponentElasticsearchMetrics, reg, path, is)
+	if err != nil {
```

I think this code was just copied, and I'm hoping I'm not contradicting what I did in other places, but I think if this is error then this can return, no need for the append.
It should be this way if there are multiple images but that isn't the case here.

---

If don't think the errors from GetReference should be handled then I don't believe GetReference should return errors. I'd rather not ignore a returned error here.

---

Sorry I wasn't clear here, I wasn't trying to suggest ignoring the error.
I was trying to say that after the `..., err = components.GetReference ...` we can just `return err`.

---

Ohh right ya this is over kill.